### ISSUE TITLE
initial review

### ISSUE THREAD

**Issue by vobencha** [RTCGA/RTCGA.CNV.20160128#1]
Hi @MarcinKosinski ,

I've done a review of RTCGA.CNV.20160128 but the comments apply to all the data packages. If you want we can just work on getting this package in shape and then you can apply changes to the others.

1) DESCRIPTION
I don't see the value in adding these fields ... is there a specific use for them?
Repository: Bioconductor
TCGAdataSet: segmented_scna_minus_germline_cnv_hg19
TCGAreleaseDate: 2016-01-28
TCGAdataType: mRNA

2) When you submit the packages for formal review we don't want files like RTCGA.CNV.20160128.Rproj in there. Can you add it to the gitignore?

3) Tags
The 'tags' for the resource metadata are taken from the biocViews field in DESCRIPTION, currently we have this:
biocViews: Annotation Data

    You may want to add a few others; valid terms found at
    http://www.bioconductor.org/packages/release/BiocViews.html#___ExperimentData
4) man page:
- The usage should show the function call with the metadata argument. The examples look good but \usage should be fixed.
- Add \format or \value section that describes the data object. Specifically, all of these stats should be explained:
```
> xx = ACC.mRNA.20160128
> class(xx)
[1] "data.frame"
> dim(xx)
[1]     5 21052
> rownames(xx)
[1] "1" "2" "3" "4" "5"
> head(colnames(xx))
[1] "bcr_patient_barcode"          "TCGA-OR-A5J1-10A-01D-A29K-01"
[3] "TCGA-OR-A5J1-10A-01D-A29K-01" "TCGA-OR-A5J1-10A-01D-A29K-01"
[5] "TCGA-OR-A5J1-10A-01D-A29K-01" "TCGA-OR-A5J1-10A-01D-A29K-01"
```
5) exporting named functions
When I type this
`ACC.mRNA.20160128`

    I should see a function definition. Instead it loads the object into the workspace. The .onLoad() is supposed to create functions with the same name as the resource so you would invoke it like this
```
> ACC.mRNA.20160128(metadata=TRUE)
Error: could not find function "ACC.mRNA.20160128"
```
    If you need an example of how to do this see the alpineData package:
```
> library(alpineData)
> ?alpineData
```
    The name of the resource is a function:
```
> ERR188297
function (metadata = FALSE)
{
    if (!isNamespaceLoaded("ExperimentHub"))
        attachNamespace("ExperimentHub")
    eh <- query(ExperimentHub(), "alpineData")
    ehid <- names(query(eh, xx))
    if (!length(ehid))
        stop(paste0("resource ", xx, "not found in ExperimentHub"))
    if (metadata)
        eh[ehid]
    else eh[[ehid]]
}
<environment: 0xb4ce4f0>
```
6) vignette:
- Not sure what date this is supposed to represent in the preamble:
date: " 17142 "
- installation section
Please remove the 'how to' install from github. Once the package is in Bioconductor it's expected to be the canonical version. This opens the door to having multiple pacakge versions at different stages of development and users get confused about where they should be downloading from. biocLite() is the suggested way so please use that.

    Remove this section:
To install this package from GitHub use
```{r, eval=FALSE}
library(RTCGA)
installTCGA("RTCGA.CNV.20160128")
```
7) valid metadata
The metadata.csv is not valid. You can test this yourself with readMetadataFromCsv():
```
xx = readMetadataFromCsv("RTCGA.CNV.20160128")
Error in readMetadataFromCsv("RTCGA.CNV.20160128") :
  missing fields in metadata.csv: Genome, SourceType, Species, TaxonomyId, Coordinate_1_based, DataProvider, DispatchClass
```
For more information see ?readMetadataFromCsv.

Valerie

**Comment by MarcinKosinski** [comment]
Hi @vobencha ! Thank you for the review :) This is repo for `RTCGA.CNV.20160128` but later you show examples for `RTCGA.mRNA.20160128` but this doesn't matter.

1. All data packages are created with `createTCGA()` function that is available only in GitHub repository where uploaded RTCGA version number is 1.5.1 - This function takes parameters specified in the DESCRIPTION file to build a data package. This is motivated by that I received many e-mails for previous data TCGA packages where people asked me which TCGA release date have I used or which data type is supported in this package or which data set name have I used to include data in this current package. Now I have all those information in 1 file, the main file for package, the DESCRIPTION file. You only specify parameters in DESCRIPTION and then you run `createTCGA` to create a data package. You can check the vull manual of `createTCGA` here - https://github.com/RTCGA/RTCGA/blob/master/R/createTCGA.R

2. I would not be so sure that you don't need this `.Rproj` file. It is added to `.Rbuildignore` anyway so it will not be build in the final package. In my opinion this file is crucial for the cooperation because it specifies parameters of the RStudio session in which the package was created: devtools options and roxygen2 options and even the encoding. 

3. This is a good idea. I may have a look at other biocViews, but I do not promise I will find appropriate biocViews for every TCGA data package.

4. I will work on `\usage` and `\value`.

5. I have copied the `.onLoad` https://github.com/RTCGA/RTCGA.mRNA.20160128/blob/master/R/zzz.R function from your vignette https://bioconductor.org/packages/release/bioc/vignettes/ExperimentHubData/inst/doc/ExperimentHubData.html#building-the-software-package . Do you want me to add extra entries to `NAMESPACE` where I would export 1 function for 1 dataset in the package with the same name as a dataset?

6. date: " 17142 " - woah! Thanks for noticing. This shoul be `as.character(Sys.Date())` instead of `Sys.Date()` here https://github.com/RTCGA/RTCGA/blob/master/R/createTCGA.R#L226 . The install from GitHub issue - I thought the Bioc-upload process might take a while that's why I left info on how to install from pre-devel source : ) It will remove that.

7. This might be the hardest. So I will work on that after the Christmas.


To sum up: I know how to improve those packages, I just need your response to 5.

**Comment by vobencha** [comment]
The ACC.mRNA.20160128 data used in the examples is from the RTCGA.CNV.20160128 package. Maybe you have the wrong files in there?

~ >ls RTCGA.CNV.20160128/data
ACC.mRNA.20160128.rda       KIPAN.mRNA.20160128.rda  PRAD.mRNA.20160128.rda
BLCA.mRNA.20160128.rda      KIRC.mRNA.20160128.rda   SARC.mRNA.20160128.rda
BRCA.mRNA.20160128.rda      KIRP.mRNA.20160128.rda   SKCM.mRNA.20160128.rda
CESC.mRNA.20160128.rda      LAML.mRNA.20160128.rda   STAD.mRNA.20160128.rda
CHOL.mRNA.20160128.rda      LIHC.mRNA.20160128.rda   STES.mRNA.20160128.rda
COADREAD.mRNA.20160128.rda  LUAD.mRNA.20160128.rda   TGCT.mRNA.20160128.rda
DLBC.mRNA.20160128.rda      LUSC.mRNA.20160128.rda   THCA.mRNA.20160128.rda
ESCA.mRNA.20160128.rda      MESO.mRNA.20160128.rda   THYM.mRNA.20160128.rda
GBMLGG.mRNA.20160128.rda    OV.mRNA.20160128.rda     UCEC.mRNA.20160128.rda
HNSC.mRNA.20160128.rda      PAAD.mRNA.20160128.rda   UCS.mRNA.20160128.rda
KICH.mRNA.20160128.rda      PCPG.mRNA.20160128.rda   UVM.mRNA.20160128.rda
    
#2: I'll follow up on this.

#5: It looks like this is related to having the data/ directory in the package so it can be ignored for now. When I remove the files in data/ (as will be the case when they're in ExperimentHub) the named functions work as expected:

```
> getNamespaceExports("RTCGA.CNV.20160128")
 [1] "UVM.mRNA.20160128"      "UCEC.mRNA.20160128"     "TGCT.mRNA.20160128"    
 [4] "CESC.mRNA.20160128"     "LIHC.mRNA.20160128"     "COADREAD.mRNA.20160128"
 [7] "ESCA.mRNA.20160128"     "KICH.mRNA.20160128"     "ACC.mRNA.20160128"     
[10] "BLCA.mRNA.20160128"     "STES.mRNA.20160128"     "BRCA.mRNA.20160128"    
[13] "LAML.mRNA.20160128"     "LUSC.mRNA.20160128"     "SKCM.mRNA.20160128"    
[16] "LUAD.mRNA.20160128"     "SARC.mRNA.20160128"     "DLBC.mRNA.20160128"    
[19] "PAAD.mRNA.20160128"     "STAD.mRNA.20160128"     "KIRC.mRNA.20160128"    
[22] "OV.mRNA.20160128"       "KIRP.mRNA.20160128"     "CHOL.mRNA.20160128"    
[25] "PRAD.mRNA.20160128"     "HNSC.mRNA.20160128"     "MESO.mRNA.20160128"    
[28] "KIPAN.mRNA.20160128"    "GBMLGG.mRNA.20160128"   "PCPG.mRNA.20160128"    
[31] "UCS.mRNA.20160128"      "THYM.mRNA.20160128"     "THCA.mRNA.20160128"    
> UVM.mRNA.20160128
function (metadata = FALSE) 
{
    if (!isNamespaceLoaded("ExperimentHub")) 
        attachNamespace("ExperimentHub")
    eh <- AnnotationHub::query(ExperimentHub(), "RTCGA.CNV.20160128")
    ehid <- names(AnnotationHub::query(eh, xx))
    if (!length(ehid)) 
        stop(paste0("resource ", xx, "not found in ExperimentHub"))
    if (metadata) 
        eh[ehid]
    else eh[[ehid]]
}
<environment: 0xa36f630>
```

8) Please remove the double colon call from .onLoad(), e.g., AnnotationHub::query. Just query() should be fine since you've imported AnnotationHub, query in the NAMESPACE. alpineData uses the same .onLoad() and does not need to use the double colon.

9) This line in .onLoad() doesn't seem to be doing anything and can be removed:
  rda <- gsub(".rda", "", titles, fixed=TRUE)
The titles (that I read in) don't have the .rda extension.

Valerie

**Comment by MarcinKosinski** [comment]
Talking about datasets' names: in the DESCRIPTION I had wrong `TCGAdataType` parameter which was set to `mRNA` instead of `CNV` :) I have changed that and rebuild this package. 

I'll follow with the rest of your comments and I will apply this to all data packages in this run. I'll let you know when I'm finished :)

**Comment by MarcinKosinski** [comment]
@vobencha I have followed your directions to improve data packages and updated `RTCGA::createTCGA` so that

- each final data package will have extended documentation with value section, this informations is also included in the readme and vignette
- there is no more information about possible installation from GitHub
- problem with incomplete metadata.csv columns is fixed
- date in the vignette is character, not a number
- onLoad function in zzz is simplified

Those changes have been applied only to the smallest package: https://github.com/RTCGA/RTCGA.mRNA.20160128 and if this package is accurate, then I will update all other packages.

I have still a problem with `usage` section in which I don't know how to specify `roxygen2` sytnax for the documentation to have `dataset(metadata = FALSE)` records in the usage, instead of just `dataset`.

Thanks for the cooperation and I wish you happy new year and a lot of energy for your future plans!

**Comment by vobencha** [comment]
@MarcinKosinski I've posted comments to the https://github.com/RTCGA/RTCGA.mRNA.20160128 repo.

**Comment by MarcinKosinski** [comment]
moved to https://github.com/RTCGA/RTCGA.mRNA.20160128/issues/1
